JOSS abstract

neuromusic opened the pull request:
This PR adds the JOSS manuscript to the repo.

closes #1440

ttung commented on paper/paper.bib:
```diff
@@ -0,0 +1,133 @@
+@Article{Seurat,
+  author = {Tim Stuart and Andrew Butler and Paul Hoffman and
+    Christoph Hafemeister and Efthymia Papalexi and William M Mauck
+    III and Yuhan Hao and Marlon Stoeckius and Peter Smibert and
+    Rahul Satija},
```
wait, they just do and and and and and, and no commas?

neuromusic replied:
yup: https://texfaq.org/FAQ-manyauthor

ttung replied:
I'd hate to be the person whose name is `and`.

neuromusic commented:
To preview the rendered PDF: http://whedon.theoj.org/

ttung commented:
http://whedon.theoj.org/preview?id=23a65e5ae3bd46f8e63a5529

mattcai commented on paper/paper.md:
```diff
@@ -0,0 +1,108 @@
+These technologies promise to permit biologists to ask new questions about the spatial relationships between cell type and interactions between gene expression and cell morphology.
+However, these assays generate very large microscopy datasets which pose challenges for data processing using standard analysis tools in microscopy. Furthermore, many of these assays require specialized analysis to decode gene expression from multiplexed experimental designs.
+
+*starfish* is a Python library for processing images generated by microscopy-based spatial transcriptomics assays. It lets biologists build scalable pipelines that localize and quantify RNA transcripts in image data generated by any FISH or In Situ Sequencing method, from classic RNA single-molecule FISH to combinatorial barcoded assays. Image processing of an experiment is divided into fields of ...
```
I think "In Situ Sequencing" should be "in situ sequencing" to refer to general methods that use sequencing rather than ISS. Or maybe "sequencing *in situ*" to really different differentiate from the specific ISS method.

mattcai replied:
```suggestion
*starfish* is a Python library for processing images generated by microscopy-based spatial transcriptomics assays. It lets biologists build scalable pipelines that localize and quantify RNA transcripts in image data generated by any FISH or sequencing *in situ* method, from classic RNA single-molecule FISH to combinatorial barcoded assays. Image processing of an experiment is divided into fields of view (FOV) that correspond to the data produced by a microscope at a single location on a microscope slide. Built on top of scikit-image and xarray, *starfish* lets users register and pre-process images in each FOV, localize spots representing tagged RNA molecules in 3D, decode the identity of those molecules according to the experimental design, segment cells, assign the spots to cells, then aggregate spots into a cell x gene expression matrix. This spatially-annotated gene expression matrix can then be analyzed in downstream tools for single-cell biology, such as Seurat (@Seurat), Bioconductor (@Bioconductor), and ScanPy (@ScanPy).
```
I think "In Situ Sequencing" should be "in situ sequencing" to refer to general methods that use sequencing. Or maybe "sequencing in situ" to really different differentiate from the specific ISS method.

mattcai commented on paper/paper.md:
```diff
@@ -0,0 +1,108 @@
+osmFISH (@osmFISH),
+BaristaSeq (@BaristaSeq),
+smFISH,
+DartFISH,
```
```suggestion
DARTFISH,
```

mattcai commented on paper/paper.md:
```diff
@@ -0,0 +1,108 @@
+BaristaSeq (@BaristaSeq),
+smFISH,
+DartFISH,
+starmap (@starmap),
```
```suggestion
STARmap (@starmap),
```

mattcai commented on paper/paper.md:
```diff
@@ -0,0 +1,108 @@
+starmap (@starmap),
+and seqFISH (@seqFISH).
+
+*starfish* is currently in use by multiple research groups, including the Allen Institute for Brain Science, the Chan Zuckerberg Biohub, and the Zhang Lab at UC San Diego. These groups support multiple large-scale projects profiling *in situ* gene expression, including the SpaceTx consortium, the Human Cell Atlas, the BRAIN Initiative Cell Census Network, and HubMap.
```
```suggestion
*starfish* is currently in use by multiple research groups, including the Allen Institute for Brain Science, the Chan Zuckerberg Biohub, and the Zhang Lab at UC San Diego. These groups support multiple large-scale projects profiling *in situ* gene expression, including the SpaceTx consortium, the Human Cell Atlas, the BRAIN Initiative Cell Census Network, and HuBMap.
```

mattcai commented on paper/paper.md:
```diff
@@ -0,0 +1,108 @@
+Marco Mignardi and Mats Nilsson for their help with the In Situ Sequencing pipeline and example data,
+Simone Codeluppi and Sten Linnarsson for their help with the osmFISH pipeline and example data,
+Xioayin Chen and Anthony Zador for their help with the BaristaSeq pipeline and example data,
+Richard Que and Kun Zhang for their help with the dartfish pipeline and example data,
```
```suggestion
Richard Que and Kun Zhang for their help with the DARTFISH pipeline and example data,
```

mattcai commented on paper/paper.md:
```diff
@@ -0,0 +1,108 @@
+Simone Codeluppi and Sten Linnarsson for their help with the osmFISH pipeline and example data,
+Xioayin Chen and Anthony Zador for their help with the BaristaSeq pipeline and example data,
+Richard Que and Kun Zhang for their help with the dartfish pipeline and example data,
+Xiao Wang, William Allen, and Karl Deisseroth for their help with the starmap pipeline and example data,
```
```suggestion
Xiao Wang, William Allen, and Karl Deisseroth for their help with the STARmap pipeline and example data,
```

kevinyamauchi commented on paper/paper.md:
```diff
@@ -0,0 +1,108 @@
+These technologies promise to permit biologists to ask new questions about the spatial relationships between cell type and interactions between gene expression and cell morphology.
+However, these assays generate very large microscopy datasets which pose challenges for data processing using standard analysis tools in microscopy. Furthermore, many of these assays require specialized analysis to decode gene expression from multiplexed experimental designs.
+
+*starfish* is a Python library for processing images generated by microscopy-based spatial transcriptomics assays. It lets biologists build scalable pipelines that localize and quantify RNA transcripts in image data generated by any FISH or sequencing *in situ* method, from classic RNA single-molecule FISH to combinatorial barcoded assays. Image processing of an experiment is divided into fields  ...
```
```suggestion
*starfish* is a Python library for processing images generated by microscopy-based spatial transcriptomics assays. Using *starfish*, biologists build scalable pipelines that localize and quantify RNA transcripts in image data generated by any FISH or sequencing *in situ* method, from classic RNA single-molecule FISH to combinatorial barcoded assays. Image processing of an experiment is divided into fields of view (FOV) that correspond to the data produced by a microscope at a single location on a microscope slide. Built on top of scikit-image and xarray, *starfish* lets users register and pre-process images in each FOV, localize spots representing tagged RNA molecules in 3D, decode the identity of those molecules according to the experimental design, segment cells, assign the spots to cells, then aggregate spots into a cell x gene expression matrix. This spatially-annotated gene expression matrix can then be analyzed in downstream tools for single-cell biology, such as Seurat (@Seurat), Bioconductor (@Bioconductor), and ScanPy (@ScanPy).
```

kevinyamauchi commented on paper/paper.md:
```diff
@@ -0,0 +1,108 @@
+STARmap (@starmap),
+and seqFISH (@seqFISH).
+
+*starfish* is currently in use by multiple research groups, including the Allen Institute for Brain Science, the Chan Zuckerberg Biohub, and the Zhang Lab at UC San Diego. These groups support multiple large-scale projects profiling *in situ* gene expression, including the SpaceTx consortium, the Human Cell Atlas, the BRAIN Initiative Cell Census Network, and HuBMap.
```
Should we provide links or references to the referenced consortia? People reading the manuscript may not know who/what they are.

neuromusic replied:
That's a good idea.

kevinyamauchi commented on paper/paper.md:
```diff
@@ -0,0 +1,108 @@
+
+*starfish* is a Python library for processing images generated by microscopy-based spatial transcriptomics assays. It lets biologists build scalable pipelines that localize and quantify RNA transcripts in image data generated by any FISH or sequencing *in situ* method, from classic RNA single-molecule FISH to combinatorial barcoded assays. Image processing of an experiment is divided into fields  ...
+
+To enable large scale processing of these data, *starfish* implements on a cloud-friendly data format that supports the 5-dimensional imaging data structures that these assays generate. Furthermore, *starfish* implements comprehensive logging of all data processing steps for provenance tracking, reproducibility, and transparency. *starfish* is built on top of popular Python tools like xarray (@xa ...
```
```suggestion
To enable large scale processing of these data, *starfish* implements on a cloud-friendly data format that supports the 5-dimensional imaging data structures (round, channel, z, y, x) that these assays generate. Furthermore, *starfish* implements comprehensive logging of all data processing steps for provenance tracking, reproducibility, and transparency. *starfish* is built on top of popular Python tools like xarray (@xarray) and scikit-image (@scikit-image).
```

kevinyamauchi commented on paper/paper.md:
```diff
@@ -0,0 +1,108 @@
+These technologies promise to permit biologists to ask new questions about the spatial relationships between cell type and interactions between gene expression and cell morphology.
+However, these assays generate very large microscopy datasets which pose challenges for data processing using standard analysis tools in microscopy. Furthermore, many of these assays require specialized analysis to decode gene expression from multiplexed experimental designs.
+
+*starfish* is a Python library for processing images generated by microscopy-based spatial transcriptomics assays. It lets biologists build scalable pipelines that localize and quantify RNA transcripts in image data generated by any FISH or sequencing *in situ* method, from classic RNA single-molecule FISH to combinatorial barcoded assays. Image processing of an experiment is divided into fields  ...
```
```suggestion
*starfish* is a Python library for processing images generated by microscopy-based spatial transcriptomics assays. It lets biologists build scalable pipelines that localize and quantify RNA transcripts in image data generated by any hybridization- or sequencing-based *in situ* transcriptomics method, from classic RNA single-molecule FISH to combinatorial barcoded assays. Image processing of an experiment is divided into fields of view (FOV) that correspond to the data produced by a microscope at a single location on a microscope slide. Built on top of scikit-image (@scikit-image) and xarray (@xarray), *starfish* lets users register and pre-process images in each FOV, localize spots representing tagged RNA molecules in 3D, decode the identity of those molecules according to the experimental design, segment cells, assign the spots to cells, then aggregate spots into a cell x gene expression matrix. This spatially-annotated gene expression matrix can then be analyzed in downstream tools for single-cell biology, such as Seurat (@Seurat), Bioconductor (@Bioconductor), and ScanPy (@ScanPy).
```

kevinyamauchi replied:
I'm not sure why the suggested changes weren't highlighted, but I just changed the second sentence ("
 It lets biologists build scalable pipelines that localize and quantify RNA transcripts in image data generated by any hybridization- or sequencing-based *in situ* transcriptomics method, from classic RNA single-molecule FISH to combinatorial barcoded assays") and added references for skimage and xarray.

kevinyamauchi commented on paper/paper.md:
```diff
@@ -0,0 +1,108 @@
+    affiliation: 2
+  - name: Tony Tung
+    affiliation: 1
+  - name: Kevin Yamauchi
```
```suggestion
  - name: Kevin A. Yamauchi
```

berl commented:
sorry to be so late to this party! need more pie. 
I do have some comments:
1. can/should we explicitly call out `slicedimage` as an interface to data in the `spacetx-format`?  These two items represent a huge chunk of work and are important contributions to the project overall.  Currently they're wrapped up in "starfish implements a cloud-friendly data format".  Not wrong, but vastly understating the utility of `spacetx-format` and `slicedimage`.
2. let's clarify that `starfish` was developed alongside the SpaceTx project, a CZI-funded effort to compare spatial transcriptomics methods in the context of determining cell types in the brain. link to  [this page](https://spacetx-starfish.readthedocs.io/en/stable/spacetx_consortium/index.html) as a "reference"?
3. does cell by gene get a shoutout as a tool for single cell biology?

neuromusic commented:
Thanks for the comments @berl... agreed on all points. I added your suggested changes. Can you review and approve?

JOSS has re-opened submissions. We also need approvals from @freeman-lab, @dganguli, and @ambrosejcarr to get this merged and submit.

ambrosejcarr commented on paper/paper.md:
```diff
@@ -0,0 +1,110 @@
+
+The exploding field of single cell transcriptomics has begun to enable deep analysis of gene expression and cell types, but spatial context is lost in the preparation of tissue for these assays.
+Recent developments in biochemistry, microfluidics, and microscopy have come together to bring about an “alphabet soup” of technologies that enable sampling gene expression in situ, with varying levels of spatial resolution, sensitivity, and genetic depth.
+These technologies promise to permit biologists to ask new questions about the spatial relationships between cell type and interactions between gene expression and cell morphology.
```
```diff
- These technologies promise to permit biologists to ask new questions about the spatial relationships between cell type and interactions between gene expression and cell morphology.
+ These technologies permit biologists to ask new questions about the spatial relationships between cell type and interactions between gene expression and cell morphology.
```
I think there are enough proofs of concept to drop the "promise to" here.

ambrosejcarr commented on paper/paper.md:
```diff
@@ -0,0 +1,110 @@
+The exploding field of single cell transcriptomics has begun to enable deep analysis of gene expression and cell types, but spatial context is lost in the preparation of tissue for these assays.
+Recent developments in biochemistry, microfluidics, and microscopy have come together to bring about an “alphabet soup” of technologies that enable sampling gene expression in situ, with varying levels of spatial resolution, sensitivity, and genetic depth.
+These technologies promise to permit biologists to ask new questions about the spatial relationships between cell type and interactions between gene expression and cell morphology.
+However, these assays generate very large microscopy datasets which pose challenges for data processing using standard analysis tools in microscopy. Furthermore, many of these assays require specialized analysis to decode gene expression from multiplexed experimental designs.
```
```diff
- However, these assays generate very large microscopy datasets which pose challenges for data processing using standard analysis tools in microscopy. Furthermore, many of these assays require specialized analysis to decode gene expression from multiplexed experimental designs.
+ However, these assays generate very large microscopy datasets which are challenging to process using general microscopy analysis tools. Furthermore, many of these assays require specialized analysis to decode gene expression from multiplexed experimental designs.
```
Suggested edit for clarity & directness.

ttung commented:
Forked as in github forked or forked as in openbsd/freebsd fork?  I like the idea of extending starfish more than the idea of forking.

ambrosejcarr commented:
> Forked as in github forked or forked as in openbsd/freebsd fork? I like the idea of extending starfish more than the idea of forking.

Extending starfish is more appealing, but I would imagine would require more investment from the team to triage/manage contributions. I think either kind of fork is better than nothing, but don't know the team's bandwidth.

neuromusic commented:
forked starfish:  https://github.com/shenlab-ucsf/SMART-Q

freeman-lab commented on paper/paper.md:
```diff
@@ -0,0 +1,116 @@
+Nico Pierson, Sheel Shah, and Long Cai for their help with the seqFISH pipeline and example data,
+and Denis Shapiro for their help with the Imaging Mass Cytometry example data.
+
+Finally, we would like to thank the SciTech team and Single Cell Program at the Chan Zuckerberg Initiative for their support of this work.
```
`SciTech` is jargon, maybe `Science and Technology teams`?

neuromusic replied:
good call.

neuromusic commented:
Thanks for the feedback, all! @dganguli gave his 👍  via email. I've addressed all of your concerns. I'm going to get this approved internally, then submit.

🎉
